.github/workflows/ngen_integration.yaml

```diff
@@ -0,0 +1,123 @@
+# Test ngen-smp integration
+
+name: Ngen Integration Tests
+
+# Controls when the action will run.
+on:
+  push:
+    branches: [ main, dev, notreal ]
+  pull_request:
+    branches: [ main, dev, notreal ]
+  workflow_dispatch:
+
+env:
+  # Obtained from https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources
+  # TODO: add something later that can check the runners for hyperthreading
+  LINUX_NUM_PROC_CORES: 2
+  MACOS_NUM_PROC_CORES: 3
+  ASAN_OPTIONS: detect_leaks=false
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # Run general unit tests in linux environment
+  test_smp:
+    # The type of runner that the job will run on
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checkout and build the SoilMoistureProfiles
+      - name: Checkout the commit
+        uses: actions/checkout@v4
+
+      - name: Build the SMP Library
+        run: |
+          cmake -B cmake_build -S . -DNGEN=ON
+          make -C cmake_build
+
+      - name: Save SoilMoistureProfiles to a Temp Directory
+        run: |
+          # Move files to a temporary directory
+          mkdir ${{runner.temp}}/smp
+          mv ./* ${{runner.temp}}/smp
+
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - name: checkout actions in another repo
+        uses: actions/checkout@v4
+        with:
+          repository: noaa-owp/soilfreezethaw
+
+      - name: Build the SFT Library
+        run: |
+          cmake -B cmake_build -S . -DNGEN=ON
+          make -C cmake_build
+
+      - name: Save SoilFreezeThaw to a Temp Directory
+        run: |
+          # Move files to a temporary directory
+          mkdir ${{runner.temp}}/sft
+          mv ./* ${{runner.temp}}/sft
```

Comment:
we don't need SoilFreezeThaw in out tests, correct?

Reply:
The relevant codes removed. We can easily add back if we want to do more complex tests in the future.